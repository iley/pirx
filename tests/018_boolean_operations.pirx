func main(): int {
  var a: int = 5;
  var b: int = 3;
  
  // Test logical AND operator with integer values (0=false, 1=true)
  printf("1 && 1: %d\n", true && true);
  printf("1 && 0: %d\n", true && false);
  printf("0 && 1: %d\n", false && true);
  printf("0 && 0: %d\n", false && false);
  
  // Test logical OR operator with integer values
  printf("1 || 1: %d\n", true || true);
  printf("1 || 0: %d\n", true || false);
  printf("0 || 1: %d\n", false || true);
  printf("0 || 0: %d\n", false || false);
  
  // Test complex expressions
  printf("(5 > 3) && (5 == 5): %d\n", (a > b) && (a == 5));
  printf("(5 < 3) || (5 == 5): %d\n", (a < b) || (a == 5));
  printf("!(5 == 3): %d\n", !(a == b));
  
  return 0;
} 
