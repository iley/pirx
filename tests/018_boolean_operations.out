1 && 1: 1
1 && 0: 0
0 && 1: 0
0 && 0: 0
1 || 1: 1
1 || 0: 1
0 || 1: 1
0 || 0: 0
(5 > 3) && (5 == 5): 1
(5 < 3) || (5 == 5): 1
!(5 == 3): 1
